src/t8_forest/t8_forest_cxx.cxx

```diff
@@ -1423,24 +1422,22 @@ t8_forest_copy_trees (t8_forest_t forest, t8_forest_t from, int copy_elements)
   }
 }
 
-/* Search for a lineasr element id (at forest->maxlevel) in a sorted array of
+/* Search for a linear element id (at forest->maxlevel) in a sorted array of
  * elements. If the element does not exist, return the largest index i
  * such that the element at position i has a smaller id than the given one.
  * If no such i exists, return -1.
  */
 /* TODO: should return t8_locidx_t */
 static t8_locidx_t
-t8_forest_bin_search_lower (t8_element_array_t *elements, t8_linearidx_t element_id, int maxlevel)
+t8_forest_bin_search_lower (const t8_element_array_t *elements, const t8_linearidx_t element_id, const int maxlevel)
```

Comment:
This is not for this PR, but we should replace the explicit binary search below by a call to std::lower_bound

Reply:
See https://github.com/DLR-AMR/t8code/issues/1089